### PR TITLE
Fix functional tests for preview

### PR DESCRIPTION
**Pull request by lfdebrux**
More fixes for G-Cloud 12

- Simplify 'I submit a copied service' step
- Fix some corner case bugs

### CODE REVIEW

File: features/step_definitions/framework_application_steps.rb [hunk 1]
```diff
@@ -149,6 +153,7 @@
 
   new_service = get_draft_service_copied_from(@existing_service, @framework['slug'])
   @new_service_href = "/suppliers/frameworks/#{@framework['slug']}/submissions/#{new_service['lotSlug']}/#{new_service['id']}"
+  puts "Draft service URL: #{@new_service_href}"
```
Comment (bjgill) [inline features/step_definitions/framework_application_steps.rb:156]: Does this need to be here?

**Review by bjgill — APPROVED** [review]
Looks sound. Just one question about an errant `puts`...

File: features/step_definitions/framework_application_steps.rb (hunk 1 above)
Comment (lfdebrux) [reply features/step_definitions/framework_application_steps.rb:156]: Oh, I put it in to trace down a bug, I thought it might be useful for future; they only show up if the test fails anyway.

Comment (bjgill) [reply features/step_definitions/framework_application_steps.rb:156]: It's just been useful - thanks! 🙂
